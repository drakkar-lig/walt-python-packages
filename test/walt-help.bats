
@test "walt help list" {
    run walt help list
    [ "$status" -eq 0 ]
}

@test "walt help show" {
    # send "q" after 2 seconds.
    # walt command should not timeout, and return code 0 (OK)
    { sleep 2; echo q; } | timeout -s INT 3 walt help show
    [ "$?" = "0" ]
}
