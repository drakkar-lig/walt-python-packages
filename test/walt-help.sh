
define_test "walt help list" as {
    walt help list
}

define_test "walt help show" as {
    # test non-interactive (no tty)
    walt help show | grep -i walt

    # test interactive session
    # we use expect to emulate a tty session.
    # we send "q" after 2 seconds.
    # walt command should not timeout, and return code 0 (OK)
    which expect || {
        echo 'This test requires the "expect" command.' >&2
        return 1
    }

    expect << EOF
set timeout 4
spawn walt help show
sleep 2
send "q\r"
expect {
    eof     { exit 0 }
    timeout { puts "timeout: 'walt help show' did not quit 4 seconds after 'q' was sent!"; exit 1 }
}
EOF
}
