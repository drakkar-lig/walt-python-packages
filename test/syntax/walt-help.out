
WalT platform control tool
-> help sub-commands

Usage:
    walt help SUBCOMMAND [args...]

Help about a given subcommand:
    walt help SUBCOMMAND --help

Help about WalT in general:
    walt help show

Sub-commands:
    list  Displays the list of help topics
    show  Displays help about a given topic
